### PR TITLE
[To rel/0.13][IOTDB-3662] upgrade audit log

### PR DESCRIPTION
**Pull request by caozj1011**
1. add master switch control;
2. add output logs to both the IOTDB and LOGGER
3. nonQuery log on PlanExecutor,query log on TsServiceIml

### CODE REVIEW

**Review by caozj1011 — COMMENTED** [review]
review

File: server/src/assembly/resources/conf/iotdb-engine.properties [hunk 1]
```diff
@@ -951,10 +951,30 @@ timestamp_precision=ms
...
 # group_by_fill_cache_size_in_mb=1.0
 
-# determines whether audit logs are written to log files or IoTDB or not write.choice:LOGGER/IOTDB/NONE
+####################
+### Audit log Configuration
+####################
+
+# whether to enable the audit log.
+# Datatype: Boolean
+#enable_audit_log=false
```
Comment (HTHou) [inline server/src/assembly/resources/conf/iotdb-engine.properties:960]:
```suggestion
# enable_audit_log=false
```

File: server/src/assembly/resources/conf/iotdb-engine.properties [hunk 2]
```diff
@@ -951,10 +951,30 @@ timestamp_precision=ms
...
+
+# Output location of audit logs
 # Datatype: String
-# audit_log_storage=NONE
+# IOTDB: the stored time series is: root.__system.audit.'user={user}'
+# LOGGER: log_audit.log in the log directory
+#audit_log_storage=IOTDB,LOGGER
```
Comment (HTHou) [inline server/src/assembly/resources/conf/iotdb-engine.properties:966]:
```suggestion
# audit_log_storage=IOTDB,LOGGER
```

File: server/src/assembly/resources/conf/iotdb-engine.properties [hunk 3]
```diff
@@ -951,10 +951,30 @@ timestamp_precision=ms
...
 
-# determines whether audit logs record IoTDB write operation
-# Datatype: boolean
-# enable_auditLog_write=false
\ No newline at end of file
+# whether enable audit log for DML operation of data
+# whether enable audit log for DDL operation of schema
+# whether enable audit log for QUERY operation of data and schema
+# Datatype: String
+#audit_log_operation=DML,DDL,QUERY
```
Comment (HTHou) [inline server/src/assembly/resources/conf/iotdb-engine.properties:972]:
```suggestion
# audit_log_operation=DML,DDL,QUERY
```

File: server/src/assembly/resources/conf/iotdb-engine.properties [hunk 4]
```diff
@@ -951,10 +951,30 @@ timestamp_precision=ms
...
+
+# whether the local write api records audit logs
+# Datatype: Boolean
+# This contains Session insert api: insertRecord(s), insertTablet(s),insertRecordsOfOneDevice
+# MQTT insert api
+# RestAPI insert api
+# This parameter will cover the DML in audit_log_operation
+#enable_audit_log_for_native_insert_api=true
```
Comment (HTHou) [inline server/src/assembly/resources/conf/iotdb-engine.properties:980]:
```suggestion
# enable_audit_log_for_native_insert_api=true
```

File: server/src/main/java/org/apache/iotdb/db/qp/physical/PhysicalPlan.java [hunk 5]
```diff
@@ -114,6 +114,8 @@ public abstract class PhysicalPlan {
    */
   private boolean isPrefixMatch = false;
 
+  protected boolean nativeInsertApi = false;
```
Comment (HTHou) [inline server/src/main/java/org/apache/iotdb/db/qp/physical/PhysicalPlan.java:117]:
```suggestion
  protected boolean isNativeInsertApi = false;
```

**Review by HTHou — APPROVED** [review]

**Pull request merged**